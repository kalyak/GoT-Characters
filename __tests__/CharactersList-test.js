import React from 'react';
import renderer from 'react-test-renderer';
import CharList from '../src/components/CharactersList';

test('renders correctly', () => {
  const tree = renderer.create(<CharList />).toJSON();
  expect(tree).toMatchSnapshot();
});
