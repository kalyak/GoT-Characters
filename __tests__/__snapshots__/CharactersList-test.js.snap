// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "aliases": Array [
            "The Daughter of the Dusk",
          ],
          "allegiances": Array [],
          "books": Array [
            "https://anapioficeandfire.com/api/books/5",
          ],
          "born": "",
          "culture": "Braavosi",
          "died": "",
          "father": "",
          "gender": "Female",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/1",
        },
        Object {
          "aliases": Array [
            "Hodor",
          ],
          "allegiances": Array [
            "https://anapioficeandfire.com/api/houses/362",
          ],
          "books": Array [
            "https://anapioficeandfire.com/api/books/1",
            "https://anapioficeandfire.com/api/books/2",
            "https://anapioficeandfire.com/api/books/3",
            "https://anapioficeandfire.com/api/books/5",
            "https://anapioficeandfire.com/api/books/8",
          ],
          "born": "",
          "culture": "",
          "died": "",
          "father": "",
          "gender": "Male",
          "mother": "",
          "name": "Walder",
          "playedBy": Array [
            "Kristian Nairn",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "Season 1",
            "Season 2",
            "Season 3",
            "Season 4",
            "Season 6",
          ],
          "url": "https://anapioficeandfire.com/api/characters/2",
        },
        Object {
          "aliases": Array [
            "Lamprey",
          ],
          "allegiances": Array [
            "https://anapioficeandfire.com/api/houses/15",
          ],
          "books": Array [
            "https://anapioficeandfire.com/api/books/3",
          ],
          "born": "",
          "culture": "",
          "died": "",
          "father": "",
          "gender": "Male",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/3",
        },
        Object {
          "aliases": Array [
            "The Merling Queen",
          ],
          "allegiances": Array [],
          "books": Array [
            "https://anapioficeandfire.com/api/books/5",
            "https://anapioficeandfire.com/api/books/8",
          ],
          "born": "",
          "culture": "Braavosi",
          "died": "",
          "father": "",
          "gender": "Female",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/4",
        },
        Object {
          "aliases": Array [
            "Old Crackbones",
          ],
          "allegiances": Array [],
          "books": Array [
            "https://anapioficeandfire.com/api/books/5",
          ],
          "born": "",
          "culture": "",
          "died": "",
          "father": "",
          "gender": "Male",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/5",
        },
        Object {
          "aliases": Array [
            "The Poetess",
          ],
          "allegiances": Array [],
          "books": Array [
            "https://anapioficeandfire.com/api/books/5",
          ],
          "born": "",
          "culture": "Braavosi",
          "died": "",
          "father": "",
          "gender": "Female",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/6",
        },
        Object {
          "aliases": Array [
            "Porridge",
          ],
          "allegiances": Array [
            "https://anapioficeandfire.com/api/houses/15",
          ],
          "books": Array [
            "https://anapioficeandfire.com/api/books/3",
          ],
          "born": "",
          "culture": "",
          "died": "",
          "father": "",
          "gender": "Female",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/7",
        },
        Object {
          "aliases": Array [
            "Quickfinger",
          ],
          "allegiances": Array [
            "https://anapioficeandfire.com/api/houses/23",
          ],
          "books": Array [
            "https://anapioficeandfire.com/api/books/6",
          ],
          "born": "",
          "culture": "",
          "died": "",
          "father": "",
          "gender": "Male",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/8",
        },
        Object {
          "aliases": Array [
            "the Sailor's Wife",
          ],
          "allegiances": Array [],
          "books": Array [
            "https://anapioficeandfire.com/api/books/5",
          ],
          "born": "",
          "culture": "",
          "died": "",
          "father": "",
          "gender": "Female",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/9",
        },
        Object {
          "aliases": Array [
            "The Veiled Lady",
          ],
          "allegiances": Array [],
          "books": Array [
            "https://anapioficeandfire.com/api/books/5",
          ],
          "born": "",
          "culture": "Braavosi",
          "died": "",
          "father": "",
          "gender": "Female",
          "mother": "",
          "name": "",
          "playedBy": Array [
            "",
          ],
          "povBooks": Array [],
          "spouse": "",
          "titles": Array [
            "",
          ],
          "tvSeries": Array [
            "",
          ],
          "url": "https://anapioficeandfire.com/api/characters/10",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              0
              :
               
              The Daughter of the Dusk
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/0/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              1
              :
               
              Walder AKA Hodor
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/1/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              2
              :
               
              Lamprey
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/2/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              3
              :
               
              The Merling Queen
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/3/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              4
              :
               
              Old Crackbones
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/4/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              5
              :
               
              The Poetess
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/5/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              6
              :
               
              Porridge
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/6/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              7
              :
               
              Quickfinger
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/7/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              8
              :
               
              the Sailor's Wife
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/8/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(230,230,230,1)",
              "height": 10,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View>
            <Text>
              9
              :
               
              The Veiled Lady
            </Text>
            <Image
              source={
                Object {
                  "uri": "https://picsum.photos/id/9/300/200",
                }
              }
              style={
                Object {
                  "height": 50,
                  "width": 50,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Text>
    Count: 
    0
  </Text>
</View>
`;
